core/src/main/kotlin/com/justai/jaicf/model/activation/selection/ContextRankingActivationSelector.kt

```diff
@@ -0,0 +1,49 @@
+package com.justai.jaicf.model.activation.selection
+
+import com.justai.jaicf.context.ActivatorContext
+import com.justai.jaicf.context.BotContext
+import com.justai.jaicf.model.activation.Activation
+import com.justai.jaicf.model.transition.Transition
+
+/**
+ * Selects the most relevant activation, based on confidence weighting.
+ *
+ * Default implementation multiplies activation confidence by a factor that is different
+ * for different layers of scenario graph. More specific transition means greater factor.
+ * Specificity is ordered as follows:
+ * 1. Children of the current state;
+ * 2. The current state itself;
+ * 3. States available from the root of scenario.
+ */
+open class ContextRankingActivationSelector : ActivationSelector {
```

Comment:
Why not to use this implementation as default selector instead of `ContextFirstActivationSelector`? Looks like it covers the context-first selector cases.